tests/IntegrationTests/InvoiceIntegrationTest.php

```diff
@@ -83,6 +84,64 @@ public function testGetInvoiceWorks()
 
         $this->assertSame('15.00', $invoice->getTotals()->getValueInc());
         $this->assertSame('15.00', $invoice->getTotals()->getValueExcl());
+
+        $this->assertNull($invoice->getFinancialNumber());
+        $this->assertNull($invoice->getFinancialCode());
+    }
+
+    public function testGetFinalInvoiceWorks()
+    {
+        $response = Response::fromString(file_get_contents(__DIR__ . '/resources/invoiceFinalGetResponse.xml'));
+
+        $this->processXmlService
+            ->expects($this->once())
+            ->method("sendDocument")
+            ->with($this->isInstanceOf(\PhpTwinfield\Request\Read\Invoice::class))
+            ->willReturn($response);
+
+        $invoice = $this->invoiceApiConnector->get('FACTUUR', '5', $this->office);
+
+        $this->assertInstanceOf(Invoice::class, $invoice);
+        $this->assertEquals(Office::fromCode("11024"), $invoice->getOffice());
+        $this->assertSame('FACTUUR', $invoice->getInvoiceType());
+        $this->assertSame('5', $invoice->getInvoiceNumber());
+        $this->assertSame('20120831', $invoice->getInvoiceDate());
+        $this->assertSame('BNK', $invoice->getBank());
+        $this->assertSame('1', $invoice->getInvoiceAddressNumber());
+        $this->assertSame('1', $invoice->getDeliverAddressNumber());
+        $this->assertSame('1000', $invoice->getCustomer()->getCode());
+        $this->assertSame('2012/8', $invoice->getPeriod());
+        $this->assertSame('EUR', $invoice->getCurrency());
+        $this->assertSame('final', $invoice->getStatus());
+        $this->assertSame('cash', $invoice->getPaymentMethod());
+
+        $invoiceLines = $invoice->getLines();
+        $this->assertCount(1, $invoiceLines);
+        $this->assertArrayHasKey('1', $invoiceLines);
+
+        /** @var InvoiceLine $invoiceLine */
+        $invoiceLine = $invoiceLines['1'];
+
+        $this->assertSame('1', $invoiceLine->getID());
+        $this->assertSame('0', $invoiceLine->getArticle());
+        $this->assertSame('118', $invoiceLine->getSubArticle());
+        $this->assertSame('1', $invoiceLine->getQuantity());
+        $this->assertSame('1', $invoiceLine->getUnits());
+        $this->assertSame('true', $invoiceLine->getAllowDiscountOrPremium());
+        $this->assertSame('CoalesceFunctioningOnImpatienceTShirt', $invoiceLine->getDescription());
+        $this->assertSame('15.00', $invoiceLine->getValueExcl());
+        $this->assertSame('0.00', $invoiceLine->getVatValue());
+        $this->assertSame('15.00', $invoiceLine->getValueInc());
+        $this->assertSame('15.00', $invoiceLine->getUnitsPriceExcl());
+        $this->assertSame('8020', $invoiceLine->getDim1());
+
+        // TODO - Vat lines
+
+        $this->assertSame('15.00', $invoice->getTotals()->getValueInc());
```

Comment:
Would you mind converting these to `Money\Money` objects, if they cannot have fractional cents?

Reply:
Hmm, as far as I can see `$invoice->getTotals()->getValueInc()` currently returns strings, the same as in the `testGetConceptInvoiceWorks`-test. Or how do you mean? :)

Reply:
I would like to change it from returning string to returning `Money` objects, this would indeed break backwards compatibility. 

We're trying to move this library (slowly) to use `Money` Value Objects instead of strings.

Reply:
That is okay, or shall I do that in another PR? 

This so you're able to release this PR without BC changes and the other one in the next BC-breaking release? It also seems not really related to this PR. In that PR I'll also be able to update the other places where `financials` are used to use the `BaseMapper`-methods.